fixed the Magento_Braintree dependency issues.

bhaveshwagento opened the pull request:
I have fixed the "Magento_Braintree" dependency issues and added the "PayPal_Braintree" module.

joeshelton-wagento commented:
Was this tested with PHP 8.1?

Also, this needs to be a revision. Change the version number in the appropriate places, update readme.md, etc.

bhaveshwagento commented:
> Was this tested with PHP 8.1?
> 
> Also, this needs to be a revision. Change the version number in the appropriate places, update readme.md, etc.

I have changed the version number and updated the readme.md. also, I have fixed the PHP 8.1 compatible issues.
I have tested the module functionality in PHP 8.1 version and it's working fine.

Please test your end if I missing any functionality. I pushed the my changes.

joeshelton-wagento commented:
I think we have to increment the version number in `composer.json`, otherwise Composer will not understand the difference between this version and the last. Right, @rkwagento ?

Why are the version numbers different between `module.xml` and `composer.json`?

bhaveshwagento commented:
> I think we have to increment the version number in `composer.json`, otherwise Composer will not understand the difference between this version and the last. Right, @rkwagento ?
> 
> Why are the version numbers different between `module.xml` and `composer.json`?

@joeshelton-wagento I think the composer.json and module.xml files version are not the same. I think We don't need to change the composer.json file as per the module README.md file.
